### PR TITLE
Dryrun v2 enhancements

### PR DESCRIPTION
**Pull request by algorandskiy**
1. Expose Dryrun input and output types in OAS
2. Add `goal clerk dryrun-remote` subcommand
3. Add `--dryrun-dump` option to commands emitting transactions
4. Add e2e expect test

### CODE REVIEW

**Comment by algorandskiy** [comment]
@algobolson @winder

File: daemon/algod/api/server/v2/account.go [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+)
+
+// AccountDataToAccount converts basics.AccountData to v2.generated.Account
+func AccountDataToAccount(
+	address string, record *basics.AccountData, assetsCreators map[basics.AssetIndex]string,
+	lastRound basics.Round, amountWithoutPendingRewards basics.MicroAlgos,
+) (generated.Account, error) {
+
+	assets := make([]generated.AssetHolding, 0)
+	if len(record.Assets) > 0 {
```
Comment (algomaxj) [inline daemon/algod/api/server/v2/account.go:34]: nit: not a necessary check

File: daemon/algod/api/server/v2/account.go [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+// AccountDataToAccount converts basics.AccountData to v2.generated.Account
+func AccountDataToAccount(
+	address string, record *basics.AccountData, assetsCreators map[basics.AssetIndex]string,
+	lastRound basics.Round, amountWithoutPendingRewards basics.MicroAlgos,
+) (generated.Account, error) {
+
+	assets := make([]generated.AssetHolding, 0)
```
Comment (algomaxj) [inline daemon/algod/api/server/v2/account.go:33]: nit: 0 is unnecessary, alternatively you could use `len(record.Assets)`

File: daemon/algod/api/server/v2/account.go [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+			asset := generated.Asset{
+				Index:  uint64(idx),
+				Params: assetParams,
+			}
+			createdAssets = append(createdAssets, asset)
+		}
+	}
+
+	var apiParticipation *generated.AccountParticipation
+	if record.VoteID != (crypto.OneTimeSignatureVerifier{}) {
```
Comment (algomaxj) [inline daemon/algod/api/server/v2/account.go:77]: Why do this check?

File: daemon/algod/api/server/v2/account.go [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+					Bytes: v.Bytes,
+					Uint:  v.Uint,
+				},
+			})
+		}
+		return
+	}
+
+	createdApps := make([]generated.Application, 0, len(record.AppParams))
+	if len(record.AppParams) > 0 {
```
Comment (algomaxj) [inline daemon/algod/api/server/v2/account.go:102]: same nit

File: daemon/algod/api/server/v2/account.go (hunk 2 above)
Comment (algorandskiy) [reply daemon/algod/api/server/v2/account.go]: done

File: daemon/algod/api/server/v2/account.go (hunk 1 above)
Comment (algorandskiy) [reply daemon/algod/api/server/v2/account.go]: done

File: daemon/algod/api/server/v2/account.go (hunk 4 above)
Comment (algorandskiy) [reply daemon/algod/api/server/v2/account.go]: done

**Pull request merged**